concept/thing/impl/ThingImpl.java

```diff
@@ -122,7 +122,7 @@ public void setHas(Attribute attribute, boolean isInferred) {
                 throw exception(TypeDBException.of(THING_KEY_OVER, attribute.getType().getLabel(), getType().getLabel()));
             } else if (attribute.getOwners(getType()).first().isPresent()) {
```

Comment:
but you can do the validation differently here:
from `Attribute.getType().getOwnersExplicit(true)` you can get the direct owners of this attribute type. Find the one that is the supertype of this `Thing` - `intersect(getOwnersExplicit(true).toSet(), superTypes().toSet())`, then use that type to check the uniqueness of the ownership with the old version of `getOwners`

Reply:
There can be many supertypes that match, and we only want the highest level one. I don't think the solution you're suggesting is any better than traversing the inheritance structure.